pkg/types/intoto/v0.0.1/entry.go

```diff
@@ -73,8 +73,24 @@ func NewEntry() types.EntryImpl {
 func (v V001Entry) IndexKeys() ([]string, error) {
 	var result []string
 
-	h := sha256.Sum256([]byte(v.env.Payload))
-	payloadKey := "sha256:" + hex.EncodeToString(h[:])
+	if v.IntotoObj.PublicKey != nil {
+		key := *v.IntotoObj.PublicKey
+		keyHash := sha256.Sum256(key)
+		result = append(result, "sha256:"+strings.ToLower(hex.EncodeToString(keyHash[:])))
+
+		pub, err := x509.NewPublicKey(bytes.NewReader(key))
+		if err != nil {
+			return result, err
+		}
+		result = append(result, pub.EmailAddresses()...)
+	}
+
+	payloadBytes, err := v.env.DecodeB64Payload()
```

Comment:
Can this be used https://github.com/sigstore/rekor/blob/ceb43c5f2edf5c6347994e017cd533f5ac0cbd3b/pkg/types/intoto/v0.0.1/entry.go#L263 for consistency?

Reply:
sure, can do that